File: algo-c-to-_/src/sum.py
#
#	from src/sum.c
#
#	float sum1(int, float[])	to	sum1
#	float sum2(int, float[])	to	sum2
#

def sum1(a):
    r = 0
    for v in a:
        r += v
    return r

def sum2(a):
    r, rest = 0, 0
    for v in a:
        t = r
        rest += v
        r += rest
        t -= r
        rest += t
    return r

File: algo-c-to-_/src/sum.lua
--
--	from src/sum.c
--
--	float sum1(int, float[])	to	sum1
--	float sum2(int, float[])	to	sum2
--

local function sum1(a)
	local r = 0
	for _,v in ipairs(a) do
		r = r + v
	end
	return r
end

local function sum2(a)
	local r, rest = 0, 0
	for _,v in ipairs(a) do
		rest = rest + v
		local t = r
		r = r + rest
		t = t - r
		rest = rest + t
	end
	return r
end

return {
	sum1 = sum1,
	sum2 = sum2
}
